kolena/_experimental/special_data_type.py

```diff
@@ -0,0 +1,81 @@
+# Copyright 2021-2024 Kolena Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Special data types supported on the Kolena platform.
+
+"""  # noqa: E501
+import os
+from abc import ABCMeta
+from datetime import datetime
+from typing import Optional
+
+from kolena._utils.datatypes import DataCategory
+from kolena._utils.datatypes import DataType
+from kolena._utils.datatypes import TypedDataObject
+from kolena._utils.pydantic_v1.dataclasses import dataclass
+from kolena._utils.validators import ValidatorConfig
+
+os.putenv("TZ", "GMT")
```

Comment:
in my understanding we only need this tz for line 80, could we set it at the func calling side like `datetime.strptime(self.value, self.format).replace(tzinfo=timezone.utc).timestamp()` for better locality?

Reply:
Yea that was my initial approach, but it turned out that `.replace()` will strip out the timezone info in the input string and lead to incorrect epoch time. In the following, this call basically just ignores the input timezone:

<img width="943" alt="image" src="https://github.com/user-attachments/assets/0d4df9ff-1be5-4062-bf8e-3622c99b58c9" />

<img width="825" alt="image" src="https://github.com/user-attachments/assets/782c23ff-d5de-46b7-ae10-e895695946a9" />

Reply:
Let me know what you think of [this commit](https://github.com/kolenaIO/kolena/pull/730/commits/c7c2cd3c76955dad17c1641bc1ab713860e2562e). I agree that manipulating `os` is bad practice and at the same time did not want to introduce new dependencies for this alone. Looking at the [python doc](https://github.com/kolenaIO/kolena/pull/730/commits/c7c2cd3c76955dad17c1641bc1ab713860e2562e):
> Changed in version 3.2: When the %z directive is provided to the [strptime()](https://docs.python.org/3/library/datetime.html#datetime.datetime.strptime) method, an aware [datetime](https://docs.python.org/3/library/datetime.html#datetime.datetime) object will be produced. The tzinfo of the result will be set to a [timezone](https://docs.python.org/3/library/datetime.html#datetime.timezone) instance.

The change is to simply check for presence of `%z` and append it to the string if not exists

Reply:
thanks for the detailed explanation, i like the changes